### PR TITLE
SDL Keyboard Input, Unix File Loading and SDL Window Config

### PR DESCRIPTION
**Pull request by megumumpkin**
Added keycode mapping from the engine's keycode enum to the appropriate SDL2 keycode enum mapping.
Unix filesystem support by changing the win32 path branching backslashes to the appropriate unix file branching slashes.
Editor with window configuration support for SDL version.

### CODE REVIEW

File: WickedEngine/wiInput.cpp [hunk 1]
```diff
@@ -548,6 +548,115 @@ namespace wiInput
                 case wiInput::KEYBOARD_BUTTON_PAGEUP:
                     keycode = SDL_SCANCODE_PAGEUP;
                     break;
+				case 48:
```
Comment (portaloffreedom) [inline WickedEngine/wiInput.cpp:551]: just one small thing, could you include in a comment a link where this codes come from? for future refence

File: WickedEngine/wiHelper.cpp [hunk 2]
```diff
@@ -382,7 +382,13 @@ namespace wiHelper
 	bool FileRead(const std::string& fileName, std::vector<uint8_t>& data)
 	{
 #ifndef PLATFORM_UWP
+#ifdef SDL_FILESYSTEM_UNIX
+		std::string unixPath = fileName;
+		std::replace(unixPath.begin(), unixPath.end(), '\\', '/');
+		ifstream file(unixPath, ios::binary | ios::ate);
+#else
```
Comment (portaloffreedom) [inline WickedEngine/wiHelper.cpp:389]: do we really need the else part? we can just do the replace in the `#ifdef SDL_FILESYSTEM_UNIX` section and load the file anyway

**Review by portaloffreedom — CHANGES_REQUESTED** [review]

File: WickedEngine/wiInput.cpp (hunk 1 above)
Comment (megumumpkin) [reply WickedEngine/wiInput.cpp:556]: Done!

**Review by portaloffreedom — DISMISSED** [review]

**Review by megumumpkin — COMMENTED** [review]
wiInput and wiHelper code has been revised

File: WickedEngine/wiHelper.cpp [hunk 3]
```diff
@@ -382,7 +382,11 @@ namespace wiHelper
 	bool FileRead(const std::string& fileName, std::vector<uint8_t>& data)
 	{
 #ifndef PLATFORM_UWP
-		ifstream file(fileName, ios::binary | ios::ate);
+		std::string filepath = fileName;
```
Comment (turanszkij) [inline WickedEngine/wiHelper.cpp:385]: There is no need to make a copy of fileName for the windows version, which doesn't modify it. So I preferred your very first solution actually.

**Comment by turanszkij** [comment]
Thanks everyone for the fixes!

**Pull request merged**